### ISSUE TITLE
Additional cursor commands

### ISSUE THREAD

**Issue by vr8hub** [Microsoft/vscode#47504]
Please consider adding "move cursor to first line on screen" and "move cursor to last line on screen" commands.  (I searched Keyboard Shortcuts and keyboard.json for "top", "screen", and "first", and can find nothing that looks like this.)

Also, consider either additional commands or (probably better) a preference that modifies the behavior of certain scroll commands. Currently, the scroll commands will move the cursor's screen position; I would like the cursor position to stay fixed on the screen.

For example, scrollLineUp moves the screen display window by one line; if lines 51-100 of the file is displayed, a scrollLineUp now displays 50-99. But if the cursor was on line 25 on the screen (line 75 in the file), the scrollLineUp moves the cursor's screen position as well; it's now on line 26 of the screen (line 75 in the file). IOW, the scroll keeps the cursor's *file* position fixed, instead of it's *screen* position. I would like the opposite.

The same for the other movements, e.g. PageUp/PageDown. Right now, in the same scenario as above, i.e. screen displays 51-100 with the cursor on line 25, a cursorPageDown only scrolls the display by 25 lines (the display is now 75-124), and the cursor is on file line 123, i.e. screen line 49. I would like the cursor position to stay fixed on screen line 25, and the file to move an entire page, i.e. 50 lines, displaying lines 101-150 after. There's no reason not to move a full page in that instance.

The only time the cursor should be moved is when the start/end of file is reached, when it's unavoidable.

In short, I would prefer VS Code not to move the cursor's screen position unless it absolutely has to. This is one thing many editors get wrong, and the editor I've used for years on Windows gets right.
